### PR TITLE
PS-776 Resolve variables request

### PR DESCRIPTION
**Pull request by pivnicek**
https://keboola.atlassian.net/browse/PS-776

### CODE REVIEW

**Comment by odinuv** [comment]
@pivnicek ^ that test should give you an idea how the API will look. If it's ok, you can start working on the data loader, if not please shout :)

**Comment by pivnicek** [comment]
was just reading it.  omg you make it look so easy :-)

**Comment by pivnicek** [comment]
I can take it from here, thanks so much

**Comment by odinuv** [comment]
No, i'll finish it; it still needs a couple of unstated modifications

**Comment by pivnicek** [comment]
ok, muchos gracias

**Comment by odinuv** [comment]
I had to change how it works - so that it works even on config rows it needs to return the entire configuration, because connection can't return state on config version, it intentionally uses empty state. It shouldn't be a problem for the sandboxes, though.

File: Tests/Controller/ApiControllerTest.php [hunk 1]
```diff
@@ -691,4 +694,357 @@ public function testProjectStats()
...
+        $this->assertArrayHasKey('version', $response);
+        $this->assertArrayHasKey('name', $response);
+        // state is always cleared
+        $this->assertEquals([], $response['state']);
+        $this->assertEquals(
+            [
+                'parameters' => [
+                    'a' => 'batman',
+                    'c' => '++++++++[>++++[>++>+++>+++>+<<<<-]>+>+>->>+[<]<-]>>.>---.+++++++..+++.>>.<-.<.+++.------.--------.>>+.>++.',
+                    'd' => 'watman',
```
Comment (MiroCillik) [inline Tests/Controller/ApiControllerTest.php:781]: watman :trollface:

File: apiary.apib [hunk 2]
```diff
@@ -417,3 +417,66 @@ Returns the sum of runtime in seconds of all jobs in the project.
...
+
+### Resolve Configuration [POST]
+
+Processes a configuration as if it were executed. This involves replacing
+[variables](https://developers.keboola.com/integrate/variables/) and
+[shared code](https://developers.keboola.com/integrate/variables/#shared-code). For configurations with rows, this also
+includes [merging configuration and configuration rows](https://help.keboola.com/components/#configuration-rows).
+The processed configuration is returned. If the configuration has rows, then the configuration itself is
+unmodified (it would not be executed). The state on both configuration and configuration rows is always returned as
+empty. The provided configuration is not unmodified, modifications are only returned.
```
Comment (MiroCillik) [inline apiary.apib:442]: Tato posledna veta mi nedava zmysel:
`The provided configuration is not unmodified, modifications are only returned.`

Nemalo by tam byt `not unmodified` -> `not modified` alebo len `unmodified`?

Comment (pivnicek) [reply apiary.apib:442]: Treba `The provided configuration has not been modified, only modifications are returned.`?

Comment (odinuv) [reply apiary.apib:442]: nj, to byl nejakej zkrat

Comment (odinuv) [reply apiary.apib:442]: dam tam `The provided configuration is not modified -- modifications are not stored, only returned` ?

Comment (pivnicek) [reply apiary.apib:442]: not sure I understand :-) .  If the provided configuration is not modified then there are no modifications to store, so the original unmodified configuration is returned?
Is that what it does?

Comment (MiroCillik) [reply apiary.apib:442]: The modified configuration is returned in the response, but it is not stored.

Comment (odinuv) [reply apiary.apib:442]: normally when you do a POST on configuration, the api changes the configuration and returns it in the response - i want to emphasize that here the api never changes the configuration

Comment (pivnicek) [reply apiary.apib:442]: I understand now, thank you.  I would put there then `The provided configuration modifications are not stored, but they are returned in the response.`

Comment (odinuv) [reply apiary.apib]: yeah, that's good :)

**Pull request merged**